addons/payment/models/payment_acquirer.py

```diff
@@ -336,6 +337,21 @@ def write(self, vals):
         self._check_required_if_provider()
         return result
 
+    def unlink(self):
+        """ Prevent the deletion of the payment acquirer if it has xmlid, because in that
+        case the record might be used in some modules.
+        """
```

Comment:
IMHO the second part is a trivial corollary.
```suggestion
        """ Prevent the deletion of the payment acquirer if it has an xmlid. """
```